org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/typesystem/computation/AbstractTypeComputer.java

```diff
@@ -41,11 +41,35 @@ protected LightweightTypeReference getTypeForName(Class<?> clazz, ITypeComputati
 		}
 		return owner.toLightweightTypeReference(type);
 	}
-	
+
+	/* @NotNull */
+	/**
+	 * @since 2.14
+	 */
+	protected LightweightTypeReference getTypeForName(String clazzName, ITypeComputationState state) {
```

Comment:
Could you please add javadoc, that clazzName is supposed to be a FQN. How should arrays look like? Same for the other overloads. Maybe simply refer to the documentation of `IJvmTypeProvider.findTypeByName(String)` or copy that snippet.